File: excelium-model/src/main/java/excelium/model/enums/StepMode.java
/*
 * MIT License
 *
 * Copyright (c) 2019 Excelium
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package excelium.model.enums;

/**
 * Represents mode of debugger execution step.
 *
 * @author PhungDucKien
 * @since 2019.12.06
 */
public enum StepMode {
    STEP_OVER,
    STEP_NEXT_BREAKPOINT,
    STEP_MUTE;

    /**
     * From name debugger execution step mode.
     *
     * @param name the name
     * @return the debugger execution step mode
     */
    public static StepMode fromName(String name) {
        for (StepMode w : StepMode.values()) {
            if (w.name().equalsIgnoreCase(name)) {
                return w;
            }
        }
        return STEP_NEXT_BREAKPOINT;
    }
}

File: packages/model/src/enums/StepMode.ts
/*
 * Licensed to the Software Freedom Conservancy (SFC) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The SFC licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

/**
 * Represents mode of debugger execution step.
 *
 * @author PhungDucKien
 * @since 2019.12.06
 */
enum StepMode {
  STEP_OVER,
  STEP_NEXT_BREAKPOINT,
  STEP_MUTE,
}

namespace StepMode {
  /**
   * From name debugger execution step mode.
   *
   * @param name the name
   * @return the debugger execution step mode
   */
  export function fromName(name: string): StepMode {
    let stepMode = (StepMode as any)[name.toUpperCase()];
    if (!stepMode) {
      stepMode = StepMode.STEP_NEXT_BREAKPOINT;
    }
    return stepMode;
  }
}

export default StepMode;
